### ISSUE TITLE
Support native module source patching

### ISSUE THREAD

**Issue by EricMCornelius** [ds300/patch-package#140]
Hi there

Really appreciate this package, has made workarounds to upstream issues much simpler in general, without needing to endlessly fork and fix minor changes that ultimately may not be adopted.

I wanted to spur some discussion on the possibility of extending the project to support patching native code updates prior to building.

This is prompted by a slew of issues I'm running into during the Node 12.x migration I'm currently experimenting with.

Seems like it would (unfortunately) be necessary to readjust the logic to copy out npm dependencies with an `--ignore-scripts` flag to initially patch the source, followed by executing the normal installation script behavior afterwards.

This, however, may be problematic, as I imagine some people are using this project to patch build output files, not pre-build sources. I don't know if it would be viable to then do a second patching pass afterwards or not.

It might all be beyond the scope you're looking for in the tool, which would be totally understandable, but I'm happy to pitch in and help if this seems like a worthwhile feature to add.

Thanks again for all the hard work!

**Comment by ds300** [comment]
> It might all be beyond the scope you're looking for in the tool

Yep, sorry :) I can see the value of being able to do that, but it's still a fairly niche use case as far as I'm aware.

If you can suggest a UX design that feels taut and easy to learn, and wouldn't affect the current UX for normal cases, then I might be open to PRs.

Otherwise I'm happy to let people keep forking for this use case 😅

**Comment by ds300** [comment]
Out of interest, what has changed in node 12 to make this feature more desirable?

**Comment by mikehardy** [comment]
I just ran into this on a real thorny dependency mess.
Specifically, a project that requires react-native 0.59.3+ (which does not support AndroidX) along with Detox (which does not support XCode 10.2 in a version that is pre-AndroidX), without wanting to downgrade to XCode <10.2. I've got just two little lines I need to change in the pre-AndroidX detox version's native XCode build, but I can't reach in there to change them before it attempts to build.

To point exactly the culprit, it's an attempt to run the CircleCI on react-native-firebase's current stable 5.x.x branch. And the change I need is a complete non-starter for merge as they've already fixed it and moved happily on, just in a version inaccessible to me in this specific setup

I have no opinion on whether this belongs in this tool or not, but I saw the issue while looking for solutions and thought a real-world use case would be interesting.

For now I suppose I'll fork detox, no huge deal

Love patch-package :-). Cheers!
